### DISCUSSION TITLE
Allow users to report name squatting, but only for data collection

### DISCUSSION THREAD

**Discussion by esoterra** [rust-lang/crates.io#4291] [Feature Requests]
To start, I think the policy of crates.io to not remove crates for name squatting is reasonable for the time being and that a policy combating name squatting would be difficult to write and implement in a way that addresses all of the potential issues.

However, I think the concern of community members that name squatting is happening and will become even more of an issue is valid.
I think the best way to address this concern without rushing into drastic changes is to start by measuring the problem and allowing users a way to report name squatting for purely data collection reasons.

This will
1. provide a data set for future analysis of this problem area
2. make users feel like they are being heard
3. give users something to do other than bug you when they find name squatting

**Comment by Turbo87** [comment]
while I appreciate the idea, I'm not sure it would be well perceived if we gathered the data without then also acting on it in some way. 

the other thought that I had was that this does not necessarily have to be built into crates.io itself and could be built on top of it.

**Reply by esoterra** [reply]
Implementing it in crates.io itself would be the most convenient for end users, because that's where they already are.
It would also be a lot harder to implement it somewhere else, since a crates.io implementation could take advantage of the existing user log in flow and a highly available UI that already has a page for each crate where a report button could be placed.

The other advantage of doing it on crates.io is that it is "official" and even if you don't actively do something about it as they come, the reports exist there for the record. It'd also give you numbers to look at to validate the assumption that it's not that big of a problem and give you an early warning if it does start to escalate.

**Comment by Turbo87** [comment]
https://crates.io/policies now restricts name squatting and we've had quite a few people reach out to our support email address to report such cases.
